aiida_common_workflows/workflows/eos.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""Equation of state workflow that can use any code plugin implementing the common relax workflow."""
+import inspect
+
+from aiida import orm
+from aiida.common import exceptions
+from aiida.engine import WorkChain, append_, calcfunction
+from aiida.plugins import WorkflowFactory
+
+from aiida_common_workflows.workflows.relax.workchain import CommonRelaxWorkChain
+
+
+def validate_inputs(value, _):
+    """Validate the entire input namespace."""
+    if 'scale_factors' not in value and ('scale_count' not in value and 'scale_count' not in value):
+        return 'neither `scale_factors` nor the pair of `scale_count` and `scale_increment` were defined.'
+
+
+def validate_sub_process_class(value, _):
+    """Validate the sub process class."""
+    try:
+        process_class = WorkflowFactory(value)
+    except exceptions.EntryPointError:
+        return f'`{value}` is not a valid or registered workflow entry point.'
+
+    if not inspect.isclass(process_class) or not issubclass(process_class, CommonRelaxWorkChain):
+        return f'`{value}` is not a subclass of the `CommonRelaxWorkChain` common workflow.'
+
+
+def validate_scale_factors(value, _):
+    """Validate the `validate_scale_factors` input."""
+    if value and len(value) < 3:
+        return 'need at least 3 scaling factors.'
+
+
+def validate_scale_count(value, _):
+    """Validate the `scale_count` input."""
+    if value is not None and value < 3:
+        return 'need at least 3 scaling factors.'
+
+
+def validate_scale_increment(value, _):
+    """Validate the `scale_increment` input."""
+    if value is not None and not 0 < value < 1:
+        return 'scale increment needs to be between 0 and 1.'
+
+
+@calcfunction
+def scale_structure(structure: orm.StructureData, scale_factor: orm.Float) -> orm.StructureData:
+    """Scale the structure with the given scaling factor."""
+    ase = structure.get_ase().copy()
+    ase.set_cell(ase.get_cell() * float(scale_factor), scale_atoms=True)
```

Comment:
I recently realized that in this way we scale the structure by the scaling factor, not the volume! Cube root of the scaling factor instead of `float(scale_factor)` would make the job.